R/mediate.R

```diff
@@ -0,0 +1,394 @@
+#' provides a simple method to analyse the effect of a treatment variable on an
+#' outcome variable found in \code{colData(se)} through the mediation of either
+#' another variable in colData (argument \code{mediator}) or an assay
+#' (argument \code{assay.type}) or a reducedDim (argument \code{dim.type}). Notably,
+#' those three arguments are mutually exclusive.
+#' 
+#' @return 
+#' A \code{data.frame} of p-values, effect sizes and confidence intervals for
+#' the ACMEs and ADEs of every mediator included in the analysis. Columns include:
+#' 
+#' \describe{
+#'   \item{Treatment}{the treatment variable}
+#'   \item{Mediator}{the mediator variable}
+#'   \item{Outcome}{the outcome variable}
+#'   \item{ACME_estimate}{the Average Causal Mediation Effect (ACME) estimate}
+#'   \item{ADE_estimate}{the Average Direct Effect (ADE) estimate}
+#'   \item{ACME_pval}{the p-value for the ACME estimate}
+#'   \item{ADE_pval}{the p-value for the ADE estimate}
+#'   \item{ACME_adjpval}{the adjusted p-value for the ACME estimate}
+#'   \item{ADE_adjpval}{the adjusted p-value for the ADE estimate}
+#'   \item{ACME_CI_lower}{the lower limit of the confidence interval for the ACME estimate}
+#'   \item{ACME_CI_upper}{the upper limit of the confidence interval for the ACME estimate}
+#'   \item{ADE_CI_lower}{the lower limit of the confidence interval for the ADE estimate}
+#'   \item{ADE_CI_upper}{the upper limit of the confidence interval for the ADE estimate}
+#' }
+#'
+#' @name getMediation
+#'
+#' @examples
+#' # Import libraries
+#' library(mia)
+#' library(microbiomeDataSets)
+#' library(scater)
+#' 
+#' # Load dataset
+#' tse <- LahtiWAData()
+#'  
+#' # Convert BMI variable to numeric
+#' tse$bmi_group <- as.numeric(tse$bmi_group)
+#' 
+#' # Analyse mediated effect of nationality on BMI through alpha diversity 
+#' med_df <- getMediation(tse,
+#'                        outcome = "bmi_group",
+#'                        treatment = "nationality",
+#'                        mediator = "diversity",
+#'                        covariates = c("sex", "age"),
+#'                        treat.value = "Scandinavia",
+#'                        control.value = "CentralEurope",
+#'                        boot = TRUE, sims = 1000)
+#' 
+#' # Apply clr transformation to counts assay
+#' tse <- transformAssay(tse,
+#'                       method = "clr",
+#'                       pseudocount = 1)
+#'
+#' # Analyse mediated effect of nationality on BMI through clr-transformed features     
+#' med_df <- getMediation(tse,
+#'                        outcome = "bmi_group",
+#'                        treatment = "nationality",
+#'                        assay.type = "clr",
+#'                        covariates = c("sex", "age"),
+#'                        treat.value = "Scandinavia",
+#'                        control.value = "CentralEurope",
+#'                        boot = TRUE, sims = 300)
```

Comment:
There are 130 features which takes long time. You could agglomerate the data to phylum level for instance to speed up

Reply:
Yes, in examples it is better to do so. But also could write in the example that it is done for the sake of the example to speed up. So that users do not robotically copypaste aggregation part and think that it is somehow necessary part of this.

Reply:
Thanks! Now it is agglomerated to family level (22 taxa) and the number of permutations was reduced to 100. I commented that we use 100 for speed but around 1000 permutations is recommended.

Reply:
Great. If it is still slow, you can just pick the top few most prevalent taxa and reduce to 50 permutations if it still works ok for a demo.